test/model-definition.test.js

```diff
@@ -391,6 +400,15 @@ describe('ModelDefinition class', function () {
     });
   });
 
+  it('should throw error for dynamic property named constructor', function(done) {
+
+    var Model = memory.createModel('DynamicCtor');
+    Model.create({ 'constructor': 'myCtor' }, function(err) {
+      should.exist(err);
```

Comment:
I think this test would pass even without your change, because we were throwing an error when trying to use `this.constructor` which was not a function. Please extend the check and verify that the messages matches the custom error.